src/main/java/org/sonar/plugins/stash/issue/collector/StashCollector.java

```diff
@@ -99,150 +98,192 @@ public static StashPullRequest extractPullRequest(String project, String reposit
-                      }
-                    }
-                  }
-                }
-              }
-            }
+    if (jsonDiffs == null) { return null; }
 
-            // Extract File Comments: this kind of comment will be attached to line 0
-            JSONArray jsonLineComments = (JSONArray) jsonDiff.get("fileComments");
-            if (jsonLineComments != null) {
+    // Let's call this for loop "objdiff_loop"
+    for (Object objDiff : jsonDiffs.toArray()) {
 
-              StashDiff initialDiff = new StashDiff(StashPlugin.CONTEXT_ISSUE_TYPE, path, 0, 0);
+      JSONObject jsonDiff = (JSONObject)objDiff;
+      // destination path in diff view
+      // if status of the file is deleted, destination == null
+      JSONObject destinationPath = (JSONObject)jsonDiff.get("destination");
 
-              for (Object objLineComment : jsonLineComments.toArray()) {
-                JSONObject jsonLineComment = (JSONObject) objLineComment;
+      if (destinationPath == null) { continue; } // Let's process the next item in "objdiff_loop"
 
-                long lineCommentId = (long) jsonLineComment.get("id");
-                String lineCommentMessage = (String) jsonLineComment.get("text");
-                long lineCommentVersion = (long) jsonLineComment.get(VERSION);
+      String path = (String)destinationPath.get("toString");
+      JSONArray jsonHunks = (JSONArray)jsonDiff.get("hunks");
 
-                JSONObject objAuthor = (JSONObject) jsonLineComment.get(AUTHOR);
-                if (objAuthor != null) {
+      if (jsonHunks == null) {
+        continue;  // Let's process the next item in "objdiff_loop"
+      }
 
-                  StashUser author = extractUser(objAuthor);
+      // Let's call this for loop "objhunk_loop"
+      for (Object objHunk : jsonHunks.toArray()) {
 
-                  StashComment comment = new StashComment(lineCommentId, lineCommentMessage, path, (long) 0, author, lineCommentVersion);
-                  initialDiff.addComment(comment);
-                }
-              }
+        JSONObject jsonHunk    = (JSONObject)objHunk;
+        JSONArray jsonSegments = (JSONArray)jsonHunk.get("segments");
 
-              result.add(initialDiff);
+        if (jsonSegments == null) {
+          continue;  // Let's process the next item in "objhunk_loop"
+        }
+
+        // Let's call this for loop "objsegm_loop"
+        for (Object objSegment : jsonSegments.toArray()) {
+
+          JSONObject jsonSegment = (JSONObject)objSegment;
+          // type of the diff in diff view
+          // We filter REMOVED type, like useless for SQ analysis
+          String type = (String)jsonSegment.get("type");
+
+          // Let's process the next item in "objsegm_loop"
+          if (StringUtils.equals(type, StashPlugin.REMOVED_ISSUE_TYPE)) {
+            continue;
+          }
+
+          JSONArray jsonLines = (JSONArray)jsonSegment.get("lines");
+
+          if (jsonLines == null) {

```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS134)